python/semantic_kernel/kernel.py

```diff
@@ -769,9 +821,13 @@ def import_semantic_plugin_from_directory(
             # Prepare lambda wrapping AI logic
             function_config = SemanticFunctionConfig(config, template)
 
-            plugin[function_name] = self.register_semantic_function(
-                plugin_directory_name, function_name, function_config
-            )
+            # TODO: this is an example of where plugins are added to the collection in the kernel
+            # as part of the register_semantic_function, seems weird to have it hidden?
+            # should the register function simply register the function and then we can add to the
+            # plugin collection later?
+            functions += [self.register_semantic_function(plugin_directory_name, function_name, function_config)]
```

Comment:
I would use `functions.extend(self.reg...` over +=

Reply:
Let me address in a subsequent PR refactor if needed.